File: homework4/src/main/java/com/example/homework4/MainActivity.java
package com.example.homework4;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Toast;

import com.google.android.material.snackbar.Snackbar;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

    private CustomView customView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        customView = findViewById(R.id.customView);
        final SharedPreferences settings = getSharedPreferences(getString(R.string.preferences_file),MODE_PRIVATE);
        customView.setTouchActionListener(new CustomView.touchActionListener() {
            @Override
            public void onTouchDown(int x, int y) {
                customView.colorChanger(x,y);
                if(settings.getBoolean("snackbar",false)){
                    Snackbar.make(findViewById(R.id.imageView),"Нажаты координаты ["+x+";"+y+"]",Snackbar.LENGTH_SHORT).show();
                }
                else{
                    Toast.makeText(MainActivity.this,"Нажаты координаты ["+x+";"+y+"]",Toast.LENGTH_SHORT).show();
                }
            }
        });
        findViewById(R.id.fab).setOnClickListener(this);
        };
    @Override
    public void onClick(View view) {
        Intent intent = new Intent(this,SettingActivity.class);
        startActivity(intent);
    }

    @Override
    protected void onSaveInstanceState(@NonNull Bundle outState) {
        super.onSaveInstanceState(outState);
        outState.putIntArray("colors",customView.getColours());
    }

    @Override
    protected void onRestoreInstanceState(@NonNull Bundle savedInstanceState) {
        super.onRestoreInstanceState(savedInstanceState);
        customView.setColours(savedInstanceState.getIntArray("colors"));
    }
}

File: homework8/src/main/java/com/example/homework8/MainActivity.kt
package com.example.homework8

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProviders
import com.example.homework8.viewmodel.MainViewModel

class MainActivity : AppCompatActivity() {
    var mainFragment = MainFragment()
    private lateinit var viewModel: MainViewModel

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        supportFragmentManager.beginTransaction()
                .add(R.id.container, mainFragment)
                .commit()

        viewModel = ViewModelProviders.of(this).get(MainViewModel::class.java)
        viewModel.title.observe(this, Observer {
            supportActionBar?.title = it
        })
        viewModel.fragment.observe(this, Observer {
            supportFragmentManager.beginTransaction().replace(R.id.container, it).addToBackStack(null).commit()
        })
    }
}
